install/linux/usr/bin/odemis-select-mic-start

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+: '
+This script is used when there are multiple modular configurations available for METEOR.
+It is recommended to combine it with the exec= command of the Odemis.desktop file.
+However it could be used stand-alone if it is required.
+The syntax to run this script -> odemis-select-mic-start [search_dir] [search_ptn].
+The script will search for configuration (.yaml) files with the used search directory
+and pattern. Once files are found microscope names are extracted and placed in a list.
+This list is presented to the user in the form of a (zenity) selection dialog window.
+After selection of the microscope configuration the right yaml file will be used to start Odemis.
+'
+# extract the arguments which were used to start this script
+args=( "$@" )
+
+for arg in ${args[@]}; do
+    if test "$arg" == "--help"; then
+        # Display Help
+        echo
+        echo "Syntax: odemis-select-mic-start [search_dir] [search_ptn]"
+        echo "arguments:"
+        echo "search_dir     The directory to search for finding the main configuration files."
+        echo "search_ptn     The search pattern to use to filter out specific file names (e.g. *meteor-main*.yaml)."
+        echo
+        exit 0
+    fi
+done
+
+# check if exactly 2 arguments are used, any other combination is invalid
+if test "${#args[@]}" != "2"; then
+        echo "Wrong number of arguments, expected 2, got ${#args[@]}"
+        echo "Syntax is: odemis-select-mic-start [search_dir] [search_ptn]"
+        exit 0
+fi
+
+# find all the files matching the search pattern and store them in an array
+files=("$(find ${args[0]} -name ${args[1]} -exec basename {} \;)")
+
+mic_list=()
+for fn in ${files[@]}; do
+    # extract the microscope name by using this small Python script
+    # note: does not check for errors when parsing
+    mic_name="$(python3 -c "import sys; import yaml; from odemis.odemisd import modelgen; f = open('${args[0]}$fn'); d = yaml.load(f, modelgen.SafeLoader); print(next(n for n, d in d.items() if d.get('class') == 'Microscope'))")"
+    mic_list+=("$fn")
+    mic_list+=("$mic_name")
+done
+
+# prompt the user with a list of microscope configurations to choose from
+selection="$(zenity --list --title "Odemis configuration starter" --list --text "Please select a configuration to start:" --column "" --column "" "${mic_list[@]}" --hide-header --width=300 --height=400 --hide-column 1 --print-column=1 2>/dev/null)"
+
+if test "$selection" != ""; then
+    odemis-start "${args[0]}$selection"
```

Comment:
I think that if the backend is already running, you should stop it first, with `sudo odemis-stop`.
Otherwise, odemis-start will detect it, and actually not start the backend again... starting the GUI for potentially a different backend.

Also, you might want at the beginning of the script check if the backend is running. In such case, you could ask the user whether to just start the GUI, or restart Odemis with a new backend. Otherwise, if the user closes the Odemis GUI window by mistake, it might be quite annoying that the only way to start it again is by restarting the whole backend. An alternative to that is to have the same behaviour as currently: if the backend is already running, just start the GUI. This can be done in the launcher with:
`if odemis check; then odemis-start; else odemis-select-mic-start; fi`